Don't use version range when declaring dependencies

lsp4xml is currently depending on lsp4j 0.5.0.M1, which uses a version range to declare its dependency towards org.eclipse.xtend.lib:

```xml
    <dependency>
      <groupId>org.eclipse.xtend</groupId>
      <artifactId>org.eclipse.xtend.lib</artifactId>
      <version>[2.11.0,3)</version>
      <scope>compile</scope>
    </dependency>
```

The Travis build [failed](https://travis-ci.org/angelozerr/lsp4xml/builds/443186891) because Travis automatically adds some snapshot repositories to its settings, and Maven then tries (and fails) to pick up a snapshot of org.eclipse.xtend.lib. 
```
[ERROR] Failed to execute goal on project org.eclipse.lsp4xml: Could not resolve dependencies for project org.eclipse:org.eclipse.lsp4xml:jar:0.0.2-SNAPSHOT: Could not find artifact org.eclipse.xtext:org.eclipse.xtext.xbase.lib:jar:2.16.0-20181005.015814-31 in sonatype-snapshots (https://oss.sonatype.org/content/repositories/snapshots/) -> [Help 1]
```

Locking down the version fixes the issue.

Using version ranges in a build is bad. Like crossing-the-streams bad (look at the NPM mess before they added lock files). You can not guarantee build idempotency so please don't do it.

--- a/build.gradle
+++ b/build.gradle
@@ -16,7 +16,7 @@ import java.time.LocalDateTime
 buildscript {
 	repositories.jcenter()
 	dependencies {
-		classpath 'org.xtext:xtext-gradle-plugin:1.0.21'
+		classpath 'org.xtext:xtext-gradle-plugin:2.0.4'
 		classpath 'io.typefox.gradle:gradle-p2gen:0.1.0'
 	}
 }

--- a/gradle/versions.gradle
+++ b/gradle/versions.gradle
@@ -13,7 +13,7 @@
 version = '0.7.0-SNAPSHOT'
 
 ext.versions = [
-	'xtext': '[2.11.0,3)',
+	'xtend_lib': '2.16.0',
 	'guava': '[14.0,22)',
 	'gson': '2.7',
 	'gson_orbit': '2.7.0.v20170129-0911',

--- a/gradle/xtend-compiler-settings.gradle
+++ b/gradle/xtend-compiler-settings.gradle
@@ -10,7 +10,7 @@
  * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
  ******************************************************************************/
 
-xtext.version = versions.xtext
+xtext.version = "[${versions.xtend_lib},3)"
 
 xtend {
 	debugger {

--- a/org.eclipse.lsp4j.generator/build.gradle
+++ b/org.eclipse.lsp4j.generator/build.gradle
@@ -15,7 +15,7 @@ description = 'Code generator for Language Server Protocol classes'
 
 dependencies {
 	compile project(":org.eclipse.lsp4j.jsonrpc")
-	compile "org.eclipse.xtend:org.eclipse.xtend.lib:$versions.xtext"
+	compile "org.eclipse.xtend:org.eclipse.xtend.lib:$versions.xtend_lib"
 }
 
 jar.manifest {

